dali/operators/nvcvop/nvcvop.cc

```diff
@@ -153,12 +153,14 @@ void AllocateImagesLike(nvcv::ImageBatchVarShape &output, const TensorList<GPUBa
 
 void PushImagesToBatch(nvcv::ImageBatchVarShape &batch, const TensorList<GPUBackend> &t_list) {
   auto channel_dim = t_list.GetLayout().find('C');
+  auto num_channels = (channel_dim >= 0) ? t_list[0].shape()[channel_dim] : 1;
+  auto format = GetImageFormat(t_list.type(), num_channels);
+  std::vector<nvcv::Image> images;
```

Comment:
We know exactly how many images there are.
```suggestion
  std::vector<nvcv::Image> images;
  images.reserve(t_list.num_samples());
```